experiment_pages/experiment/map_rfid.py

```
@@ -160,30 +160,31 @@ def listen():
                     last_rfid = received_rfid
                     print(f"📡 RFID Scanned: {received_rfid}")
 
+                    # ✅ Clean up garbage chars and whitespace
                     clean_rfid = received_rfid.strip().replace("\x02", "").replace("\x03", "")
 
                     if not clean_rfid:
                         print("⚠️ Empty or invalid RFID detected, skipping...")
-                        continue  # 🚫 Avoid calling add_value()
+                        continue
 
-                    if clean_rfid in self.animal_rfid_list:
-                        print(f"⚠️ RFID {clean_rfid} is already in use! Skipping...")
-                        play_sound_async("shared/sounds/error.wav")
-                        continue  # 🚫 Avoid calling add_value()
+                    animal_id = local_db.get_animal_id(clean_rfid)
+                    if animal_id is not None:
+                        print(f"✅ Found Animal ID: {animal_id}")
+                        self.after(0, lambda: self.select_animal_by_id(animal_id))
+                    else:
+                        print("❌ No animal found for scanned RFID:", clean_rfid)

```

This should also raise an error, rather than just printing one within the terminal.